### ISSUE TITLE
googletest update

### ISSUE THREAD

**Issue by johnomotani** [boutproject/BOUT-dev#1638]
After #1630 went in, the unit tests failed to compile for me because googletest did not automatically update itself to the new version. Doing:
```
git submodule update --init --recursive
```
fixed the problem for me. Is there a nicer solution?

**Comment by johnomotani** [comment]
Checking further, the `rm` and `git checkout` that I had in my original post were not necessary, just `git submodule update --init --recursive` is enough to update googletest. Would be nice if this could be automatic or we could print some kind of helpful warning? Otherwise, this issue gives the workaround for anyone who has this problem.

PS the error messages when trying to compile tests with the old version of google test started like this:
```
field/test_field_factory.cxx:70:17: error: expected constructor, destructor, or type conversion before ‘(’ token
 TYPED_TEST_SUITE(FieldFactoryCreationTest, Fields);
                 ^
In file included from ../../externalpackages/googletest/googletest/include/gtest/gtest.h:66:0,
                 from field/test_field_factory.cxx:1:
field/test_field_factory.cxx:72:1: error: ‘gtest_type_params_FieldFactoryCreationTest_’ was not declared in this scope
 TYPED_TEST(FieldFactoryCreationTest, CreateFromValueGenerator) {
 ^
field/test_field_factory.cxx:72:1: error: template argument 3 is invalid
 TYPED_TEST(FieldFactoryCreationTest, CreateFromValueGenerator) {
```

**Comment by ZedThree** [comment]
I'm not sure how automatic we can make this as it mostly depends on what version of git you have, how you cloned the repo, and your personal git settings.

I think most submodule problems can be solved by running:

    git config --global submodule.recurse true

which then makes the update automatic, but this requires git 2.14.

**Comment by ZedThree** [comment]
This is very likely "can't/won't fix", see: https://stackoverflow.com/a/1899826/2043465
Another answer to the same question: https://stackoverflow.com/a/43854593/2043465

Someone else on a different question suggested making an alias for `git checkout && git submodule update --init` for earlier versions of git.

In order to make this automatic for us, we'd likely need to keep track of the submodule commit id separately from git, and I don't fancy that. We might like to add something to the developer's documentation though.

**Comment by johnomotani** [comment]
:+1: the `git config` option fixes this issue for me. Would be good to add it to the developer documentation.
